tools/shared.py

```diff
@@ -1896,8 +1896,9 @@ def link_lld(args, target, opts=[], lto_level=0):
       cmd += [
         '-z', 'stack-size=%s' % Settings.TOTAL_STACK,
         '--initial-memory=%d' % Settings.TOTAL_MEMORY,
-        '--no-entry'
       ]
+      if not Settings.STANDALONE_WASM or '_main' not in Settings.EXPORTED_FUNCTIONS:
+        cmd += ['--no-entry']
```

Comment:
this line seems to assume that if standalone then we always have a `main()`? I'm not sure we want that. Wasi is limited to that so far, but standalone != wasi.

Reply:
This line is saying that "if STANDALONE_WASM and we have main then we look for _start".    If there is no main or if we are not in STANDALONE_WASM mode then _start doesn't need to exist.  So I think its doing what we (you) want here.

I could rewrite it as:

```
use_start = Settings.STANDALONE_WASM and  '_main'  in Settings.EXPORTED_FUNCTIONS
if not use_start:
  cmd += ['--no-entry']
```

Reply:
Thanks, yes, that version looks clearer.